### PR TITLE
soc: nordic: nRF54L move DCDC configuration to DT

### PR DESCRIPTION
**Pull request by kl-cruz**
It is follow up of this PR: https://github.com/zephyrproject-rtos/zephyr/pull/74592
Includes nRF54L15 device

### CODE REVIEW

File: soc/nordic/nrf54l/soc.c [hunk 1]
```diff
@@ -149,7 +149,13 @@ static int nordicsemi_nrf54l_init(void)
 		nrf_power_task_trigger(NRF_POWER, NRF_POWER_TASK_CONSTLAT);
 	}
 
-	if (IS_ENABLED(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)) {
+#if (DT_PROP(DT_INST(0, nordic_nrf5x_regulator), regulator_initial_mode) == \
+		NRF5X_REG_MODE_DCDC)
+	{
+#else
+	/* DCDC is disabled in DTS. Do not configure it. */
+	if (0) {
```
Comment (gmarull) [inline soc/nordic/nrf54l/soc.c:157]: why is this needed?

**Review by gmarull — CHANGES_REQUESTED** [review]
please make changes preserving bisectability

File: soc/nordic/nrf54l/soc.c (hunk 1 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/soc.c]: updated

File: soc/nordic/nrf54l/soc.c [hunk 2]
```diff
@@ -149,9 +149,10 @@ static int nordicsemi_nrf54l_init(void)
 		nrf_power_task_trigger(NRF_POWER, NRF_POWER_TASK_CONSTLAT);
 	}
 
-	if (IS_ENABLED(CONFIG_SOC_NRF54L_VREG_MAIN_DCDC)) {
-		nrf_regulators_vreg_enable_set(NRF_REGULATORS, NRF_REGULATORS_VREG_MAIN, true);
-	}
+#if (DT_PROP(DT_INST(0, nordic_nrf5x_regulator), regulator_initial_mode) == \
```
Comment (gmarull) [inline soc/nordic/nrf54l/soc.c:152]:
```suggestion
#if (DT_PROP(DT_NODELABEL(vregmain), regulator_initial_mode) == \
```

**Review by gmarull — COMMENTED** [review]
pending: squash, current set is not bisectable

File: dts/common/nordic/nrf54l15.dtsi [hunk 3]
```diff
@@ -577,6 +578,13 @@
 				interrupts = <270 NRF_DEFAULT_IRQ_PRIORITY>;
 				status = "disabled";
 			};
+
+			reg: regulator@120600 {
+				compatible = "nordic,nrf5x-regulator";
+				reg = <0x120600 0x1>;
+				regulator-name = "REG";
+				regulator-initial-mode = <NRF5X_REG_MODE_DCDC>;
+			};
```
Comment (gmarull) [inline dts/common/nordic/nrf54l15.dtsi:587]: this seems wrong: there are 2 regulators that are children of this IP block: VREGM, and VREGMAIN

File: soc/nordic/nrf54l/soc.c (hunk 2 above)
Comment (kl-cruz) [reply soc/nordic/nrf54l/soc.c]: done

File: dts/common/nordic/nrf54l15.dtsi (hunk 3 above)
Comment (kl-cruz) [reply dts/common/nordic/nrf54l15.dtsi]: As adviced: Added group for regulators, new dts description for nrf54x regulators, commits squashed

File: dts/bindings/regulator/nordic,nrf54x-regulators.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Nordic REGULATORS (voltage regulators control module) on nRF54X.
+
+compatible: "nordic,nrf54x-regulators"
```
Comment (gmarull) [inline dts/bindings/regulator/nordic,nrf54x-regulators.yaml:6]:
```suggestion
compatible: "nordic,nrf54l-regulators"
```

Comment (kl-cruz) [reply dts/bindings/regulator/nordic,nrf54x-regulators.yaml]: Changed names to 54l

File: dts/common/nordic/nrf54l15.dtsi [hunk 5]
```diff
@@ -577,6 +578,22 @@
...
+				compatible = "nordic,nrf54l-regulators";
+				reg = <0x120000 0x1000>;
+				status = "okay";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				vregmain: regulator@120600 {
+					compatible = "nordic,nrf5x-regulator";
+					reg = <0x120600 0x1>;
+					status = "okay";
```
Comment (gmarull) [inline dts/common/nordic/nrf54l15.dtsi:592]: this should be enabled at board level, see other families

Comment (gmarull) [reply dts/common/nordic/nrf54l15.dtsi:592]: is non-DCDC an option, right?

Comment (kl-cruz) [reply dts/common/nordic/nrf54l15.dtsi:592]: yes, it is also an option

Comment (kl-cruz) [reply dts/common/nordic/nrf54l15.dtsi]: I moved enabling and DCDC configuration to board level

**Review by gmarull — COMMENTED** [review]
one last request: commit title is not clear, should be something like `soc: nordic: nrf54l: move DCDC configuration to DT`

**Comment by gmarull** [comment]
commit title still missing nrf54l..

**Review by gmarull — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Comment by gmarull** [comment]
assigned to myself, as I now maintain regulators and did similar changes for 52/53.

**Review by fabiobaltieri — APPROVED** [review]
